UrlManager.php

```diff
@@ -197,7 +197,14 @@ public function createUrl($params)
             if (!$languageRequired && !$this->enableDefaultLanguageUrlCode && $language===$this->getDefaultLanguage()) {
                 return  $url;
             } else {
-                $url = rtrim($url, '/');
+                foreach ($this->rules as $rule) {
+                    if(in_array($rule->route, $params)) {
+                        if($rule->name=='' && $rule->suffix!='/') {
+                            $url = rtrim($url, '/');
+                        }
+                        break;
+                    }
+                }
```

Comment:
Hmm, I'd prefer if we find a solution that avoids looping over all rules each time. And I wonder if this solution is perfect anyway. What about home URLs created like `site/index`? Maybe I miss the point, though. I still had no time to rethink the problem.